compiler/parser/src/fstring.rs

```diff
@@ -6,63 +6,72 @@ use crate::{
 };
 use std::{iter, mem, str};
 
-struct FStringParser {
+struct FStringParser<'a> {
+    chars: iter::Peekable<str::Chars<'a>>,
     str_start: Location,
     str_end: Location,
 }
 
-impl FStringParser {
-    fn new(str_start: Location, str_end: Location) -> Self {
-        Self { str_start, str_end }
+impl<'a> FStringParser<'a> {
+    fn new(source: &'a str, str_start: Location, str_end: Location) -> Self {
+        Self {
+            chars: source.chars().peekable(),
+            str_start,
+            str_end,
+        }
+    }
+
+    fn next_char(&mut self) -> Option<char> {
+        self.chars.next()
+    }
+
+    fn peek(&mut self) -> Option<&char> {
+        self.chars.peek()
     }
```

Comment:
@DimitrisJim @youknowone Probably not necessary, but should these two functions be `#[inline]`?

Reply:
I'd say so, afaik the compiler usually inlines small functions like these but that isn't guaranteed, hinting it with `inline` is helps.

Reply:
I'd like to have inline for tiny functions.